go/cmd/dolt/commands/reset.go

```diff
@@ -80,90 +83,111 @@ func (cmd ResetCmd) ArgParser() *argparser.ArgParser {
 		return HandleVErrAndExitCode(verr, usage)
 	} else if apr.Contains(HardResetParam) {
-		return handleResetHard(ctx, apr, usage, dEnv, roots)
-	} else {
-		if apr.NArg() == 1 {
-			ref := apr.Arg(0)
-			isValidRef, err := actions.IsValidRef(ctx, ref, dEnv.DoltDB, dEnv.RepoStateReader())
-			if err != nil {
-				return handleErrAndExit(err)
-			}
-			if isValidRef {
-				return handleResetSoftToRef(ctx, dEnv, ref, usage)
-			}
+		if apr.NArg() > 1 {
+			return handleResetError(fmt.Errorf("--hard supports at most one additional param"), usage)
 		}
-
-		return handleResetSoftTables(ctx, apr.Args, roots, dEnv, usage)
 	}
-}
 
-func handleResetSoftTables(ctx context.Context, tables []string, roots doltdb.Roots, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
-	roots, err := actions.ResetSoft(ctx, dEnv.DbData(), tables, roots)
+	// process query through prepared statement to prevent sql injection
+	query, params := constructParametrizedDoltResetQuery(apr)
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
 	if err != nil {
-		return handleResetError(err, usage)
+		cli.Println(err.Error())
+		return 1
 	}
-
-	err = dEnv.UpdateRoots(ctx, roots)
+	_, _, err = queryist.Query(sqlCtx, interpolatedQuery)
 	if err != nil {
-		return handleResetError(err, usage)
+		cli.Println(err.Error())
+		return 1
 	}
 
-	printNotStaged(ctx, dEnv, roots.Staged)
-	return 0
-}
-
-func handleResetSoftToRef(ctx context.Context, dEnv *env.DoltEnv, ref string, usage cli.UsagePrinter) int {
-	newRoots, err := actions.ResetSoftToRef(ctx, dEnv.DbData(), ref)
-	if err != nil {
-		return handleResetError(err, usage)
+	if !apr.Contains(HardResetParam) {
+		printNotStaged(sqlCtx, queryist)
 	}
 
-	err = dEnv.UpdateStagedRoot(ctx, newRoots.Staged)
-	return handleResetError(err, usage)
+	return 0
 }
 
-func handleResetHard(ctx context.Context, apr *argparser.ArgParseResults, usage cli.UsagePrinter, dEnv *env.DoltEnv, roots doltdb.Roots) int {
-	arg := ""
-	if apr.NArg() > 1 {
-		return handleResetError(fmt.Errorf("--hard supports at most one additional param"), usage)
-	} else if apr.NArg() == 1 {
-		arg = apr.Arg(0)
-	}
+// constructParametrizedDoltResetQuery generates the sql query necessary to call the DOLT_RESET() stored procedure with placeholders
+// for arg input. Also returns a list of the inputs in the order in which they appear in the query.
+func constructParametrizedDoltResetQuery(apr *argparser.ArgParseResults) (string, []interface{}) {
+	var params []interface{}
+	var param bool
 
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	ws, err := dEnv.WorkingSet(ctx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	var buffer bytes.Buffer
+	var first bool
+	first = true
+	buffer.WriteString("CALL DOLT_RESET(")
+
+	writeToBuffer := func(s string) {
```

Comment:
This pattern is becoming familiar. Are there some functions we can move into common shared functions? I'm not going to block on a refactor for this, but there may be a way to simplify this and reuse code